src/splitBy.js

```diff
@@ -0,0 +1,24 @@
+var _curry2 = require('./internal/_curry2');
+var complement = require('./complement');
+var difference = require('./difference');
+var takeWhile = require('./takeWhile');
+
+
+/**
+ * Splits a given list at the first occurance of a satisfied predicate.
+ *
+ * @func
+ * @memberOf R
+ * @category List
+ * @sig Number -> [a] -> [[a], [a]]
+ * @param {Function} fn The predicate that determines where the array is split.
+ * @param {Array} array The array to be split.
+ * @return {Array}
+ * @example
+ *
+ *      R.splitBy(R.equals(2), [1, 2, 3]);   //=> [[1], [2, 3]]
+ */
+module.exports = _curry2(function splitBy(predicate, array) {
+  var xs = takeWhile(complement(predicate), array);
+  return [xs, difference(array, xs)];
```

Comment:
We can't use `difference` here, as it discards duplicates. Try something like `R.splitBy(R.T, [1, 1, 1])`.

It would be great to use [JSVerify][1] to test these properties:

  - `forall p :: a -> Boolean, xs :: [a]. length(splitBy(p, xs)) = 2`
  - `forall p :: a -> Boolean, xs :: [a]. unnest(splitBy(p, xs)) = xs`


[1]: https://github.com/jsverify/jsverify